### PR TITLE
added workaround for table indexes in globals.lua

### PR DESCRIPTION
**Pull request by actionless**
for some reason (i thing because of some compatibility layer with vim) table index starts from 0 instead of 1 (as normally in lua)

so vim produced args:
{0: 'file/path.lua', 1: 0 }

while lua code was waiting for smth like:
{1: 'file/path.lua', 2: 0 }

### CODE REVIEW

File: misc/lua-ftplugin/globals.lua [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env lua
 
 -- Parse output of "luac -l" for GETGLOBAL/SETGLOBAL instructions.
-local command = string.format('luac -p -l "%s"', arg[1])
-local verbose = tonumber(arg[2]) ~= 0
+local command = string.format('luac -p -l "%s"', arg[0])
```
Comment (actionless) [inline misc/lua-ftplugin/globals.lua:4]: also it will be good to use path to luac from settings instead of hardcoded one

**Comment by xolox** [comment]
I think I just resolved both of the issues you reported. I did so in a different way than you suggested because I wanted to fix the problem at the root cause (the integration with the Lua Interface for Vim) so that I won't have to use the same workaround in multiple scripts.

I'm closing this pull request now because I believe the issue to be resolved, but if it's not then please feel free to reopen the pull request. Thanks for the feedback!

**Comment by actionless** [comment]
good, gonna check it :)

**Comment by xolox** [comment]
Finally ran into this issue myself today, after building Vim from source and editing a Lua script. Turns out the issue wasn't properly fixed yet! I was able to read the updated `if_lua.txt` documentation now though, so I understood the cause of the problem and how to solve it once and for all. This is now properly fixed in 4746f4d4bddb51619c2956d73adfb22b73e066dd.

**Pull request closed without merging**